components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/persistence/impl/HybridCertificateValidationPersistenceManager.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.x509Certificate.validation.persistence.impl;
+
+import static org.wso2.carbon.identity.x509Certificate.validation.constant.error.ErrorMessage.ERROR_CERTIFICATE_DOES_NOT_EXIST;
+import static org.wso2.carbon.identity.x509Certificate.validation.constant.error.ErrorMessage.ERROR_INVALID_VALIDATOR_NAME;
+import static org.wso2.carbon.identity.x509Certificate.validation.constant.error.ErrorMessage.ERROR_NO_VALIDATORS_CONFIGURED_ON_TENANT;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import org.wso2.carbon.identity.x509Certificate.validation.exception.CertificateValidationManagementException;
+import org.wso2.carbon.identity.x509Certificate.validation.model.CACertificate;
+import org.wso2.carbon.identity.x509Certificate.validation.model.CACertificateInfo;
+import org.wso2.carbon.identity.x509Certificate.validation.model.Validator;
+import org.wso2.carbon.identity.x509Certificate.validation.persistence.CertificateValidationPersistenceManager;
+
+/**
+ * HybridCertificateValidationPersistenceManager is a hybrid implementation of CertificateValidationPersistenceManager
+ * which uses both JDBC and Registry based persistence managers.
+ */
+public class HybridCertificateValidationPersistenceManager implements CertificateValidationPersistenceManager {
+
+    private final JDBCCertificateValidationPersistenceManager jdbcCertificateValidationPersistenceManager =
+            new JDBCCertificateValidationPersistenceManager();
+    private final RegistryCertificateValidationPersistenceManager registryCertificateValidationPersistenceManager =
+            new RegistryCertificateValidationPersistenceManager();
+
+    @Override
+    public void addValidators(List<Validator> validators, String tenatDomain)
+            throws CertificateValidationManagementException {
+
+        jdbcCertificateValidationPersistenceManager.addValidators(validators, tenatDomain);
+    }
+
+    @Override
+    public void addCACertificates(List<Validator> validators, List<X509Certificate> trustedCertificates,
+                                  String tenantDomain) throws CertificateValidationManagementException {
+
+        jdbcCertificateValidationPersistenceManager.addCACertificates(validators, trustedCertificates, tenantDomain);
```

Comment:
Same https://github.com/wso2-extensions/identity-x509-commons/pull/59/files#r1948382017 applies to this as well.

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/59/commits/ea4895653a265439feb6fba55276aca051cf319f